sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Rules/Hotspots/DoNotHardcodeCredentialsBase.cs

```diff
@@ -118,20 +124,23 @@ protected CredentialWordsFinderBase(DoNotHardcodeCredentialsBase<TSyntaxKind> an
                 context =>
                 {
                     var declarator = (TSyntaxNode)context.Node;
-
                     if (!IsAssignedWithStringLiteral(declarator, context.SemanticModel))
                     {
                         return;
                     }
-
                     var variableName = GetVariableName(declarator);
                     var variableValue = GetAssignedValue(declarator);
 
                     var bannedWords = FindCredentialWords(variableName, variableValue);
                     if (bannedWords.Any())
                     {
-                        context.ReportDiagnosticWhenActive(
-                            Diagnostic.Create(rule, declarator.GetLocation(), string.Join(", ", bannedWords)));
+                        context.ReportDiagnosticWhenActive(Diagnostic.Create(rule, declarator.GetLocation()
+                            , string.Format(MessageFormatCredential, bannedWords.JoinStr(", ", x => $@"""{x}"""))));
```

Comment:
Could we extract the `"` character inside the message format to use `x => $"{x}"` here instead.
Or remove the `@` string from the message format since it is not currently needed

Reply:
I've moved it into message. It will produce `"a, b, c" detected here` instead of `"a", "b", "c" detected here`. It's not so nice, but it matches the RSPEC message.